src/expression.rs

```diff
@@ -30,11 +30,16 @@ impl<TS: TypeSystem> Debug for NativeFunction<TS> {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum VariableType {
```

Comment:
what about heap values, do we have a plan for how users can handle them? there should be a comment here at least mentioning that later task.

Reply:
I don't understand what you're asking, heap values are always going to have a reference kept on the stack, the Rc in the InternalReference is what's making it heap allocated. Freight itself doesn't explicitly have a section of memory called heap, that's handled by rust's allocator.